core/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -84,6 +85,29 @@ public TransportSearchAction(Settings settings, ThreadPool threadPool, Transport
         return aliasFilterMap;
     }
 
+    private Map<String, Float> resolveIndexBoosts(SearchRequest searchRequest, ClusterState clusterState) {
+        if (searchRequest.source() == null) {
+            return Collections.emptyMap();
+        }
+
+        SearchSourceBuilder source = searchRequest.source();
+        if (source.indexBoosts() == null) {
+            return Collections.emptyMap();
+        }
+
+        Map<String, Float> concreteIndexBoosts = new HashMap<>();
+        for (SearchSourceBuilder.IndexBoost ib : source.indexBoosts()) {
+            String[] concreteIndices =
+                indexNameExpressionResolver.concreteIndexNames(clusterState, searchRequest.indicesOptions(), ib.getIndex());
```

Comment:
Can we switch to using the method that returns `Index[]` and use the index uuid rather than the index name as key in the map? That will help with the upcoming cross cluster search feature ;)